### ISSUE TITLE
[BUG] Failed sync on chain

### ISSUE THREAD

**Issue by thisUsernameAlreadyExists** [sunflower-land/sunflower-land#377]
BUG:  Failed sync on chain.

BUG DESCRIPTION: 
1. Syncing on chain caused an issue with gas fee payment. 
2. After refresh another issue popped but only after the gas fee payment and while in process of sync.3. 

Even though it failed. After refresh the inventory was restocked and I was able to do a successful sync.

DEVICE USED: Macbook Pro 2015

BROWSER USED: Firefox developer edition 99.0b3 (64-bit)

BUG REPLICATED: No

NOTES: I usually open the website and enter the game plant some seeds. Then close the tab but the web browser is till opened. Since after the alarm sound on my phone pops I go back in and set another round of seeds.

I tried to replicate this but doesn't happen. Maybe it's something that has to do with the session

This is for the first issue (1) from bug description:
![Screenshot 2022-03-15 at 14 34 32](https://user-images.githubusercontent.com/34162331/158387350-fc050055-dfb7-434f-9a44-cee927ff1614.png)
![Screenshot 2022-03-15 at 14 35 53](https://user-images.githubusercontent.com/34162331/158387325-7401acd1-ad7d-402e-b82c-343821ac43e3.png)

For second issue (2) from bug description:
![Screenshot 2022-03-15 at 14 47 36](https://user-images.githubusercontent.com/34162331/158387522-51da8220-abad-428b-825a-8fd5de702d43.png)
![Screenshot 2022-03-15 at 14 46 35](https://user-images.githubusercontent.com/34162331/158387523-f88700c2-e6e2-46be-82eb-b3fc8b8a0f06.png)

I'll come back if theres anyway to replicate it.
Hope this helps a bit!

**Comment by adamhannigan** [comment]
@thisUsernameAlreadyExists There have been a range of improvements to fix the syncing bug over the last 3-4 days.

Are you still experiencing this issue?

**Comment by thisUsernameAlreadyExists** [comment]
@adamhannigan I didn't encounter it anymore in the past few days.
Looks good to me now.
